### ISSUE TITLE
Doesn't 'tf.py_func' layers need tf.RegisterGradient for gradients?

### ISSUE THREAD

**Issue by bareblackfoot** [endernewton/tf-faster-rcnn#288]
For example, proposal layer, which is regressed to the proposal target layer,the gradient from it should be calculated and go back through the "rpn_cls_score" layer etc.. But if the network doesn't have 'register gradient', I think the gradient would stop. Could anyone explain me about this?

**Comment by endernewton** [comment]
you are right, so that means faster rcnn is NOT end-to-end. however, in the paper they did show the network is not trained stably when back proped to the image coordinates for example.
